### ISSUE TITLE
Site Usability: Apisix versions are easy mistaken

### ISSUE THREAD

**Issue by anhldbk** [iresty/iresty.github.io#1469]
### Current State

**Description of the current state/issue**:

I like the modern design system of our website.

I think there's an issue with our web usability: the version block and the component one are too far from each other.

![web-issue](https://user-images.githubusercontent.com/3270746/211341257-f2683704-ef49-4446-a2bf-00965027942e.png)

Users may mistakenly think this is the version of apisix instead.

**Link to the page**:

Any doc page, for example: https://apisix.apache.org/docs/apisix/getting-started/

### Desired State

**Description of the change you are proposing**:

We can move the component-selector block to the sidebar at the top-right corner

### Would you like to fix this issue?

No. Please assign another contributor

**Comment by SkyeYoung** [comment]
We are preparing to modify this part. I have had no time recently, but I will help to modify this part as soon as possible.

@juzhiyuan @guoqqqi Maybe you have time to help improve this?

**Comment by juzhiyuan** [comment]
No, there have a lot of TODOs... Just added good first issue

**Comment by RajVarsani** [comment]
I would like to work on this issue

**Comment by SkyeYoung** [comment]
@RajVarsani 😸 Great! If there are any problems, please feel free to @ any of us. (Hope you won't be dissuaded by the old ugly code. 🥲)

**Comment by RajVarsani** [comment]
Sure
I'll start with setup then, will update here if I encounter any issue.

**Comment by anhldbk** [comment]
@SkyeYoung Can we use the same versions for sub components?

**Comment by SkyeYoung** [comment]
@anhldbk Could you please give me an example? I'm not sure if I'm thinking what you're talking about. 🤔

**Comment by anhldbk** [comment]
@SkyeYoung AFAIK Apisix comprises of various sub components with differrent versions in releases. Multiple versions lead to this issue and other compatibility issues like [this one](https://github.com/apache/apisix-ingress-controller/issues/1435#issuecomment-1374964779).

My question is: can we use the same version for each component? [Vertx](https://github.com/eclipse-vertx/vert.x) uses this approach.

**Comment by RajVarsani** [comment]
@SkyeYoung, I set up the codebase in my locally. I have encountered a couple of issues.
I opened it in the Dev container, and it shows 624 (Which is the total number of files in the repo ig) file changes because of End of Line Difference (I'm in windows). 
![image](https://user-images.githubusercontent.com/74860406/215487529-d11bbfd5-1347-4a93-b9dd-20fd37dba654.png)
I'm using this command: `yarn start:doc` . How do I see changes reflected in the server? I tried changing the content of the nav links, but it is not reflecting as hot or even hard reload. I had need to restart the server to see the changes

**Comment by RajVarsani** [comment]
@SkyeYoung, I set up the codebase in my locally. I have encountered a couple of issues.
I opened it in the Dev container, and it shows 624 (Which is the total number of files in the repo ig) file changes because of End of Line Difference (I'm in windows).
![image](https://user-images.githubusercontent.com/74860406/215494090-b3d18579-d7b4-4aed-9e42-28c1209e6675.png)

I'm using this command: `yarn start:doc` . How do I see changes reflected in the server? I tried changing the content of the nav links, but it is not reflecting as hot or even hard reload. I had need to restart the server to see the changes

**Comment by SkyeYoung** [comment]
@RajVarsani 

For the end-of-line symbol problem, I think you can fix it by clicking `Ctrl + Shift + P` and typing `Change End of Line Sequence` and selecting `LF`.

![image](https://user-images.githubusercontent.com/48400568/215498441-19b05bf1-7ccf-40d7-8749-8fe235bdb5be.png)

For the question about hot reloading, it's not really supposed to have this problem ... I need to try it out. But I don't use Windows directly for any development work, including installing docker-desktop on it, so it will take a bit of time and I may be able to tell you tomorrow

**Comment by RajVarsani** [comment]
Update: I've discarded all previous changes and updated vs code prefs, so the LF-CRLF issue seems to be resolved. Thanks, @SkyeYoung, Lmk if you find any update on the reloading issue.

**Comment by SkyeYoung** [comment]
> @SkyeYoung AFAIK Apisix comprises of various sub components with differrent versions in releases. Multiple versions lead to this issue and other compatibility issues like [this one](https://github.com/apache/apisix-ingress-controller/issues/1435#issuecomment-1374964779).
> 
> My question is: can we use the same version for each component? [Vertx](https://github.com/eclipse-vertx/vert.x) uses this approach.

🤔 cc @juzhiyuan

**Comment by juzhiyuan** [comment]
There has no plan to release packages with the same version numbers. But I suggest adding some notes on each project's documentation: which version is compatible?

A better way is let each component to auto detect version 🤔

**Comment by SkyeYoung** [comment]
> Update: I've discarded all previous changes and updated vs code prefs, so the LF-CRLF issue seems to be resolved. Thanks, @SkyeYoung, Lmk if you find any update on the reloading issue.

@RajVarsani Sorry for too late. I tried it, but it doesn't seem to reproduce the thermal reloading problem. I tried searching and found this https://stackoverflow.com/questions/65445600/hot-reload-is-not-working-in-my-react-app. I don't know if it will help you...

**Comment by RajVarsani** [comment]
I'll check and update here ✅

**Comment by Mackenzie-OO7** [comment]
@SkyeYoung @anhldbk Hello, can I work on this issue?

**Comment by kondapalli19** [comment]
Hey, I'm new to open source and would like to work on this so can you please assign me this issue.

**Comment by SkyeYoung** [comment]
@Mackenzie-OO7 @kondapalli19 Hi, I think you can try to collaborate and work together to complete it. Alternatively, you can try dividing the work among yourselves and each person can complete their own part. This will require you to collaborate and communicate with each other. 😸

**Comment by kondapalli19** [comment]
@SkyeYoung T component-selector block you want to put that at the top of the version selector block or something else

**Comment by kondapalli19** [comment]
@Mackenzie-OO7 Hi! Thank you for the suggestions. I think dividing the work among ourselves and communicating with each other would be a good approach. Let's discuss how we can divide the work and make sure we're on the same page. Looking forward to working together on this! 😊

**Comment by rukundob451** [comment]
Hey @anhldbk, is this still available?

**Comment by anhldbk** [comment]
@rukundob451 make a PR fast pls

**Comment by aarishshahmohsin** [comment]
Can I be assigned this issue?

**Comment by Rishab87** [comment]
Hi  , Can I work on this issue?

**Comment by HattySohaib** [comment]
@SkyeYoung Hi, it seems like this issue was never resolved. Can I work on this?

**Comment by rohitvkgdg** [comment]
Hey @SkyeYoung @juzhiyuan I am new to open-source, can I work on this issue?

**Comment by SkyeYoung** [comment]
> Hey [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/48400568?u=72ff53ea9b7dc2c6aa26b21a20af4f2eb8c255b4&amp;v=4&amp;size=80">@SkyeYoung](https://github.com/SkyeYoung?rgh-link-date=2025-02-09T19%3A28%3A57.000Z) [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/2106987?u=5b04797d5dc7a1d955dc46f3e2c1bb57c2abbdb0&amp;v=4&amp;size=80">@juzhiyuan](https://github.com/juzhiyuan?rgh-link-date=2025-02-09T19%3A28%3A57.000Z) I am new to open-source, can I work on this issue?

@rohitvkgdg Of course, feel free to do so.(Sorry I haven't been paying attention to this project lately.)

**Comment by Devnil434** [comment]
Please make me assigned in this issue as I want to start to my open source contribution

**Comment by juzhiyuan** [comment]
Hi @Devnil434, @rohitvkgdg has taken this one

**Comment by yzeng25** [comment]
Closing this for now due to inactivity, but feel free to reopen if this is still relevant.
